Added functionalities

subdesign opened the pull request:
I added the following functions:
- optional email activation
- user registration with reCaptcha
- forgot password 
- alert message email to admin if user is locked out (optional)

Commit:
Expanded functionality

Commit:
removed a file

danmontgomery commented:
I dig this. Need to find time in the next few days to look it over, and I'll merge it in.

subdesign commented:
All right, feel free to modify or prettify the code if needed.

CroNiX commented:
I'm trying to test this but getting the "To use reCAPTCHA you must get an API key from https://www.google.com/recaptcha/admin/create" message.  I do have a key entered, but I'm testing this on localhost and google states any key should work on localhost.

Commit:
Modified code based on CroNIX comments

Commit:
Added date helper for using mdate()
Added email config option 'mailtype'
Added set_value('username') for login view

Commit:
Added modification based CroNIX comments

Commit:
Merge branch 'master' of github.com:subdesign/codeigniter-bitauth

subdesign commented:
I don't clearly understand the problem with reCaptcha, can you explain in a bit more detail?

CroNiX commented:
Call to undefined method Bitauth::get_user_by_email()
Called by Bitauth::generate_forgot_code()

CroNiX commented:
@subdesign I was just testing the base install with your mod, set the recaptcha keys in the config (even though I'm testing on localhost/127.0.0.1 and google says any key will work on localhost), and got the error that I reported above.  I've disabled it as I don't need it for what I'm doing, but it didn't work for my tests.

ehabafia commented on application/views/example/add_user.php:
```diff
@@ -15,6 +15,7 @@
 	<title>BitAuth: <?php echo $title; ?></title>
 </head>
 <body>
+<script>
```
Why is this <script> here, I think this should be moved out.
